lib/presentation/pages/qvst/widgets/response_reference_widget.dart

```diff
@@ -22,32 +25,41 @@ class ResponseReferenceWidget extends ConsumerStatefulWidget {
 class _ResponseReferenceWidgetState
     extends ConsumerState<ResponseReferenceWidget> {
   QvstAnswerRepoEntity? _selectedRepo;
+  QvstAnswerRepoEntity? _initialRepo;
+  bool _isSaved = true;
+
+  void _initializeRepo(List<QvstAnswerRepoEntity> data) {
+    if (_initialRepo == null) {
+      _initialRepo =
+          data.firstWhere((element) => element.id == widget.referenceId);
+      _selectedRepo = _initialRepo;
+    }
+  }
```

Comment:
prefer use provider arguments instead

see [Passing arguments to provider](https://riverpod.dev/fr/docs/essentials/passing_args)

Should be

```dart
final selectedRepo = ref.watch(qvstAnswerRepoListProvider(widget.referenceId));
``` 

init data when you receive other data is not recommended and can cause poor performances on UI